src/Controller/SiteInfoController.php

```diff
@@ -0,0 +1,203 @@
+   *   The bundle name.
+   *
+   * @return mixed
+   *   Returns all required fields for bundle.
+   */
+  protected function getRequiredFieldWidgets($entity_type, $bundle) {
+    $entity_form_display = EntityFormDisplay::load("{$entity_type}.{$bundle}.default");
+    $form_display_widgets = $entity_form_display->getComponents();
+
+    $field_definitions = $this->entityFieldManager->getFieldDefinitions($entity_type, $bundle);
+
+    $required_fields = array_reduce(
+      array_keys($form_display_widgets),
+      function ($collection, $field_name) use ($form_display_widgets, $field_definitions) {
+        $field_info = $form_display_widgets[$field_name];
+        if (!isset($field_info['region']) || $field_info['region'] !== 'content') {
+          return $collection;
+        }
+
+        if (isset($field_definitions[$field_name]) && $field_definitions[$field_name]->isRequired()) {
+          $collection[$field_name] = $field_info;
+        }
+
+        return $collection;
+      },
+      []
+    );
+
+    return $required_fields;
+  }
+
+  /**
+   * Get bundles ordered by number of instances.
+   *
+   * @param array $data
+   *   The collected data from sampler module for entity type.
+   *
+   * @return array
+   *   Returns bundle names ordered by count.
+   */
+  protected function getBundlesByCount(array $data) {
+    $bundles_by_count = [];
+
+    foreach ($data as $bundle_name => $bundle_info) {
+      if (!isset($bundle_info['instances'])) {
+        continue;
+      }
+
+      $bundles_by_count[$bundle_name] = $bundle_info['instances'];
+    }
+
+    arsort($bundles_by_count);
+
+    return $bundles_by_count;
+  }
+
+  /**
+   * Get bundles ordered by number of instances.
+   *
+   * @param array $data
+   *   The collected data from sampler module for entity type.
+   *
+   * @return array
+   *   Returns bundle names ordered by count.
+   */
+  protected function getBundlesByNumberOfFields(array $data) {
+    $bundles_by_number_of_fields = [];
+
+    foreach ($data as $bundle_name => $bundle_info) {
+      $bundles_by_number_of_fields[$bundle_name] = count(array_keys($bundle_info['fields']));
+    }
+    arsort($bundles_by_number_of_fields);
+
+    return $bundles_by_number_of_fields;
+  }
+
+  /**
+   * Gets site information.
+   *
+   * Allowed query options are following:
+   * => rule: {string} - options:
+   *   - count: to sort bundles by number of instance
+   *   - number_of_fields: to sort bundles by number of fields
+   * Default: "count"
+   *
+   * => index: {int}
+   * Default: 0
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   The symfony request object.
+   *
+   * @return \Symfony\Component\HttpFoundation\JsonResponse
+   *   Returns JSON with site information needed by performance tests.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function siteInfo(Request $request) {
+    $rule = $request->query->get('rule', 'count');
```

Comment:
In the current implementation, nothing could go wrong with strange, values for the rule parameter. 
But we should still provide a sanity check, if one of the expected values are given.